src/program_generator/pg_marginal.py

```diff
@@ -89,18 +91,82 @@ def _do_generate(self, opcode, max_op_count, shuffle_counts, step_op_count):
       
-    programs = [self._generate_single_program(operation, op_count) for operation in operations for op_count in op_counts]
+    programs = [self._generate_single_program(operation, op_count, max_op_count) for operation in operations for op_count in op_counts]
 
     return programs
 
-  def _generate_single_program(self, operation, op_count):
+  def _generate_single_program(self, operation, op_count, max_op_count):
+    assert op_count <= constants.MAX_INSTRUCTIONS
     # for compatibility with the generate_single_marginal function
     if operation['Mnemonic'] == 'CREATE':
-        return Program(generate_create_program(operation, op_count), operation['Mnemonic'], op_count)
-
+      return Program(_generate_create_program(operation, op_count, max_op_count), operation['Mnemonic'], op_count)
+    if operation['Mnemonic'] in ['EXTCODEHASH', 'EXTCODESIZE']:
+      return Program(_generate_extcode_program(operation, op_count, max_op_count), operation['Mnemonic'], op_count)
+    if operation['Mnemonic'] == 'EXTCODECOPY':
+      return Program(_generate_extcodecopy_program(operation, op_count, max_op_count), operation['Mnemonic'], op_count)
+    if operation['Mnemonic'] in ['CALL', 'STATICCALL', 'DELEGATECALL']:
+      return Program(_generate_call_program(operation, op_count, max_op_count), operation['Mnemonic'], op_count)
+  
     single_op_pushes = ["6003"] * arity(operation)
 
     return Program(generate_single_marginal(single_op_pushes, operation, op_count), operation['Mnemonic'], op_count)
+
+def _generate_create_program(create_operation, op_count, max_op_count):
+  """
+  Generates a program for CREATE opcode
+  """
+  assert create_operation['Mnemonic'] == 'CREATE'
+
+  account_code = '6d6460016001016000526005601cf3600052'
+  empty_pushes = '6000' * constants.MAX_INSTRUCTIONS
+  single_op_pushes = '600d60126000' * op_count
+  opcodes_with_pops = (create_operation['Value'][2:4] + '50') * op_count
+  pops = '50' * (constants.MAX_INSTRUCTIONS - op_count)
+  return account_code + empty_pushes + single_op_pushes + opcodes_with_pops + pops
+
+def _generate_extcode_program(create_operation, op_count, max_op_count):
+  """
+  Generates a program for EXTCODEHASH and EXTCODESIZE opcodes
+  """
+  assert create_operation['Mnemonic'] == 'EXTCODEHASH' or create_operation['Mnemonic'] == 'EXTCODESIZE'
+
+  empty_pushes = '60ff' * constants.MAX_INSTRUCTIONS
+  account_code = '6c63ffffffff60005260046000f3600052600d60006000f0'
+  opcode_args = '80' * (max_op_count - 1) # duplicate the address on stack
+  opcodes_with_pops = (create_operation['Value'][2:4] + '50') * op_count
+  pops = '50' * (constants.MAX_INSTRUCTIONS + max_op_count - op_count)
+  return empty_pushes + account_code + opcode_args + opcodes_with_pops + pops
+
+def _generate_extcodecopy_program(create_operation, op_count, max_op_count):
+  """
+  Generates a program for EXTCODECOPY opcode
+  """
+  assert create_operation['Mnemonic'] == 'EXTCODECOPY'
+
+  empty_pushes = '60ff' * 5
```

Comment:
Differs from most other opcodes. Usually we stacked 60 x 'ff', but in the programs we have ran we put only `ff` on stack only 5 times.